### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by fiesh**
Fix gcc11 compilation issues and update and de-bloat `OpenSceneGraph` along the way.

### CODE REVIEW

**Comment by fiesh** [comment]
Oh there's a typo in the first commit message, `gcc1` instead of `gcc11`.  If this is approved, I can fix it.

**Comment by fiesh** [comment]
To be honest, I'd really like to remove all optional dependencies from `OpenSceneGraph` by default... All of `dcmtk` `gdal` `giflib` `gstreamer` `gta` `jasper` `jpeg` `libpng` `openal` `openexr` `poppler` `tiff` `zlib` are optional and lead to a lot of bloat while probably not being used by a typical `OpenSceneGraph` user.

**Comment by fiesh** [comment]
It might actually make more sense to track `OpenSceneGraph` directly on github than via its releases... I'd be happy to change it.

**Comment by tonytheodore** [comment]
> I'd really like to remove all optional dependencies from OpenSceneGraph by default

We leave as much enabled as possible since it's easier for users to disable what they don't need than try to enable new features. I've just [added PKG_CMAKE_OPTS](https://github.com/mxe/mxe/commit/97beb1d5c7ad962fdd4df496085a6d918bb21230) to mimic `PKG_CONFIGURE_OPTS`, can you add the following:

```
diff --git a/src/openscenegraph.mk b/src/openscenegraph.mk
index f05a90ae..d740d268 100644
--- a/src/openscenegraph.mk
+++ b/src/openscenegraph.mk
@@ -27,7 +27,8 @@ define $(PKG)_BUILD
         -DDYNAMIC_OPENTHREADS=$(CMAKE_SHARED_BOOL) \
         -DDYNAMIC_OPENSCENEGRAPH=$(CMAKE_SHARED_BOOL) \
         -DBUILD_OSG_APPLICATIONS=OFF \
-        -DOPENTHREADS_ATOMIC_USE_MUTEX=ON
+        -DOPENTHREADS_ATOMIC_USE_MUTEX=ON \
+        $(PKG_CMAKE_OPTS)

     $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)' VERBOSE=1
     $(MAKE) -C '$(BUILD_DIR)' -j 1 install VERBOSE=1
```
Then users can:
```
make openscenegraph openscenegraph_DEPS=cc openscenegraph_CMAKE_OPTS='-DENABLE_FOO=OFF -DENABLE_BAR=OFF'
```

The other two changes look fine, and I'm happy to switch to tracking github tags/releases.

**Comment by fiesh** [comment]
> > I'd really like to remove all optional dependencies from OpenSceneGraph by default
> 
> We leave as much enabled as possible since it's easier for users to disable what they don't need than try to enable new features. I've just [added PKG_CMAKE_OPTS](https://github.com/mxe/mxe/commit/97beb1d5c7ad962fdd4df496085a6d918bb21230) to mimic `PKG_CONFIGURE_OPTS`, can you add the following:
> 
> ```
> diff --git a/src/openscenegraph.mk b/src/openscenegraph.mk
> index f05a90ae..d740d268 100644
> --- a/src/openscenegraph.mk
> +++ b/src/openscenegraph.mk
> @@ -27,7 +27,8 @@ define $(PKG)_BUILD
>          -DDYNAMIC_OPENTHREADS=$(CMAKE_SHARED_BOOL) \
>          -DDYNAMIC_OPENSCENEGRAPH=$(CMAKE_SHARED_BOOL) \
>          -DBUILD_OSG_APPLICATIONS=OFF \
> -        -DOPENTHREADS_ATOMIC_USE_MUTEX=ON
> +        -DOPENTHREADS_ATOMIC_USE_MUTEX=ON \
> +        $(PKG_CMAKE_OPTS)
> 
>      $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)' VERBOSE=1
>      $(MAKE) -C '$(BUILD_DIR)' -j 1 install VERBOSE=1
> ```
> 
> Then users can:
> 
> ```
> make openscenegraph openscenegraph_DEPS=cc openscenegraph_CMAKE_OPTS='-DENABLE_FOO=OFF -DENABLE_BAR=OFF'
> ```

Ok perfect... what do we do about (at least with `gcc10` and `gcc11`) broken `gdal`?  Leave it in nonetheless?

> The other two changes look fine, and I'm happy to switch to tracking github tags/releases.

I'll switch, thanks!

**Comment by tonytheodore** [comment]
Until gdal is updated:
```
make openscenegraph gdal_CONFIGURE_OPTS='--with-openjpeg=no --with-cryptopp=no'
```

**Comment by fiesh** [comment]
So it was already based on github... I meant to track specific revisions of `master`, but I guess that's too hard to follow for people, releases are more meaningful.  So I think this would be good to go now.

**Comment by fiesh** [comment]
Is there anything left to do for this to be merged?

**Comment by tonytheodore** [comment]
Just reset the _DEPS back to the original.

Note if you are tracking a repo locally, you can do:
```
make openscenegraph openscenegraph_SOURCE_TREE=/path/to/local/repo
```
and it will build against whatever your working set is at the time.

**Comment by fiesh** [comment]
Ok, should be ok now I hope!  Thanks for the helpful insight regarding the build directory.

**Comment by tonytheodore** [comment]
Thanks!
